packages/mask/src/plugins/Tips/contexts/Tip/useAddNameService.ts

```diff
@@ -0,0 +1,29 @@
+import { useChainId, useWeb3State } from '@masknet/plugin-infra/web3'
+import { useEffect, useMemo, useState } from 'react'
+import type { AddressConfig } from '../../types'
+
+export function useAddNameService(addressConfigs: AddressConfig[]) {
+    const [map, setMap] = useState<Record<string, string>>({})
+
+    const { NameService } = useWeb3State()
+    const chainId = useChainId()
+    useEffect(() => {
```

Comment:
`useAsync`

Reply:
`async-await` is not in the top level of `useEffect` callback

Reply:
So, we can use `await Promise.all()`.